src/components/templates/PartyMap/components/Map/Map.tsx

```diff
@@ -56,28 +52,122 @@ interface MapProps {
+ *
+ * B = the browser window.
+ *
+ *            #--------------BBBBBBBBBBBBBBBBBBBBBBB--------------#
+ *            |  Total image B Visible portion     B              |
+ *            |              B inside browser      B              |
+ *            |              B                     B              |
+ *            |              B                     B              |
+ *            |              B                     B              |
+ *            |              B---------------------B              |
+ *            |              B                     B              |
+ *            |              B Area where portals  B              |
+ *            |              B appear.             B              |
+ *            |              B                     B              |
+ *            |              B                     B              |
+ *            |              B                     B              |
+ *            |              B                     B              |
+ *            |              B                     B              |
+ *            |              B---------------------B              |
+ *            |              B                     B              |
+ *            |              B                     B              |
+ *            |              B                     B              |
+ *            |              B                     B              |
+ *            #--------------BBBBBBBBBBBBBBBBBBBBBBB--------------#
+ *
+ * The central area is called the "safe zone". This must always be visible.
+ * Portals are positioned so that they appear to be fixed to the image.
+ */
 export const Map: React.FC<MapProps> = ({ user, venue, selectRoom }) => {
-  const [mapBackground] = useValidImage(
-    venue?.mapBackgroundImageUrl,
-    DEFAULT_MAP_BACKGROUND
-  );
+  const [
+    mapBackground,
+    { width: imageWidth, height: imageHeight },
+  ] = useValidImage(venue?.mapBackgroundImageUrl, DEFAULT_MAP_BACKGROUND);
+
+  const { width: _browserWidth, height: browserHeight } = useWindowSize();
+
+  const browserWidth = useMemo(() => {
+    return document.body.clientWidth;
+    // This is intentinonally recalculated when the browser resizes
+    // This allows for optional horizontal scrollbar sizes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [_browserWidth]);
 
   if (!user || !venue) {
     return <>Loading map...</>;
   }
 
+  if (!imageWidth || !imageHeight) {
+    // TODO Tidy thisup
+    console.error("Failed to get image width/height");
+    return <>Error</>;
+  }
+
+  const { width: safeWidthPerc, height: safeHeightPerc } = venue.config
+    ?.safeZone || {
+    width: 1.0,
+    height: 1.0,
+  };
+
+  // TODO Account for height of top and bottom nav
+
+  const verticalPad = 140; // 70px for top and bottom
+
+  // Figure out if the image should be rescaled using the width or height as
+  // a basis
+  const safeZoneWidth = safeWidthPerc * imageWidth;
```

Comment:
[NOTE] Generally the naming here can be made more self-documenting. Some ideas:
- `scale` or `factor` for `perc`
- prepending or appending `old`, `new` or `candidate` to the names instead of underscore
- `zone`, instead of `safeZone`, to keep them shorter since there isn't other type of zone